airflow/upgrade/rules/import_changes.py

```diff
@@ -0,0 +1,89 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import NamedTuple, Optional, List
+
+from cached_property import cached_property
+
+from airflow import conf
+from airflow.upgrade.rules.base_rule import BaseRule
+from airflow.upgrade.rules.renamed_classes import ALL
+from airflow.utils.dag_processing import list_py_file_paths
+
+
+class ImportChange(
+    NamedTuple(
+        "ImportChange",
+        [("old_path", str), ("new_path", str), ("providers_package", Optional[None])],
+    )
+):
```

Comment:
can be simplified

```suggestion
class ImportChange(NamedTuple)
    old_path: str
    new_path: str
    providers_package: Optional[None]

```

Reply:
oh wait, I know why you did this, for compatibility with Py 2.7 ?

Reply:
Yup, not a big fan of this but we this command has to work with 2.7

Reply:
Some operators did the other way.. from providers to core
https://github.com/apache/airflow/pull/10013
this message will be wrong for it?

Reply:
The message will be:
> Using `airflow.operators.email_operator.EmailOperator` will be replaced by `airflow.operators.email.EmailOperator`.

And I see no place for confusion here